images/linux/scripts/installers/java-tools.sh

```diff
@@ -7,34 +7,55 @@ source $HELPER_SCRIPTS/install.sh
 source $HELPER_SCRIPTS/os.sh
 source $HELPER_SCRIPTS/etc-environment.sh
 
-JAVA_VERSIONS_LIST=$(get_toolset_value '.java.versions | .[]')
-DEFAULT_JDK_VERSION=$(get_toolset_value '.java.default')
-
-# Install GPG Key for Adopt Open JDK. See https://adoptopenjdk.net/installation.html
-wget -qO - "https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public" | apt-key add -
-add-apt-repository --yes https://adoptopenjdk.jfrog.io/adoptopenjdk/deb/
-
-if isUbuntu16 || isUbuntu18 ; then
-    # Install GPG Key for Azul Open JDK. See https://www.azul.com/downloads/azure-only/zulu/
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 0xB1998361219BD9C9
-    apt-add-repository "deb https://repos.azul.com/azure-only/zulu/apt stable main"
-fi
-apt-get update
-
-for JAVA_VERSION in ${JAVA_VERSIONS_LIST[@]}; do
-    apt-get -y install adoptopenjdk-$JAVA_VERSION-hotspot=\*
-    echo "JAVA_HOME_${JAVA_VERSION}_X64=/usr/lib/jvm/adoptopenjdk-${JAVA_VERSION}-hotspot-amd64" | tee -a /etc/environment
-done
+JAVA_TOOLCACHE_PATH="$AGENT_TOOLSDIRECTORY/Java_Adoptium_jdk"
+
+createEnvironmentVariable() {
+    local JAVA_VERSION=$1
+    local JAVA_PATH=$2
+
+    local JAVA_HOME_PATH=$JAVA_PATH/Contents/Home
+    if [[ $JAVA_VERSION == $JAVA_DEFAULT ]]; then
+        echo "JAVA_HOME=${JAVA_HOME_PATH}" | tee -a /etc/environment
+    fi
+
+    echo "JAVA_HOME_${JAVA_VERSION}_X64=${JAVA_HOME_PATH}" | tee -a /etc/environment
+}
+
+installJavaFromAdoptOpenJDK() {
+    local JAVA_VERSION=$1
+
+    javaRelease=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/${JAVA_VERSION}/hotspot" \
```

Comment:
Do we need take a first index in the array?
`jq -r '.[] | select(.binary.os=="linux" and .binary.image_type=="jdk" and .binary.architecture=="x64")`

Reply:
Thank you, got rid of unnecessary `[0]`